examples/eps_to_csv/resources/csv_merger.py

```diff
@@ -0,0 +1,243 @@
+            num_rows_in_original_target = len(target_df)
+        except pd.errors.EmptyDataError:
+            logger.info(f"Target CSV '{target_csv_path}' exists but is empty. Will treat as new.")
+            target_df = pd.DataFrame() # Start with empty DataFrame
+            original_target_df_existed = False
+        except Exception as e:
+            logger.error(f"Error reading target CSV '{target_csv_path}': {e}", exc_info=True)
+            raise
+    else:
+        logger.info(f"Target CSV '{target_csv_path}' not found or is empty. Will create it.")
+        target_df = pd.DataFrame() # Start with empty DataFrame
+
+    # Identify new cluster_names from source_df
+    source_cluster_names = set(source_df['cluster_name'].dropna().unique())
+
+    if not target_df.empty and "cluster_name" in target_df.columns:
+        target_cluster_names = set(target_df['cluster_name'].dropna().unique())
+    else:
+        target_cluster_names = set()
+        if not target_df.empty and "cluster_name" not in target_df.columns:
+            logger.warning(f"Target CSV '{target_csv_path}' exists but lacks 'cluster_name' column. All source entries will be treated as new if they have 'cluster_name'.")
+
+    new_cluster_names_to_add = source_cluster_names - target_cluster_names
+
+    new_rows_list = []
+    if new_cluster_names_to_add:
+        logger.info(f"Identified {len(new_cluster_names_to_add)} new cluster name(s) to add from source.")
+        for cluster_name_to_add in new_cluster_names_to_add:
+            # Get the first row matching the cluster_name (in case of duplicates in source)
+            source_row_series = source_df[source_df['cluster_name'] == cluster_name_to_add]
+            if source_row_series.empty:
+                # Should not happen if new_cluster_names_to_add was derived correctly
+                logger.warning(f"Cluster name '{cluster_name_to_add}' was in new_cluster_names_to_add but not found in source_df during iteration. Skipping.")
+                continue
+            source_row = source_row_series.iloc[0]
+            
+            new_row_data = {}
+
+            # Populate based on EXPECTED_TARGET_COLUMNS
+            new_row_data['cluster_name'] = source_row['cluster_name']
+            new_row_data['cluster_group'] = source_row.get('cluster_group', pd.NA)
+            new_row_data['cluster_tags'] = source_row.get('cluster_tags', pd.NA)
+
+            new_row_data['platform_repository_revision'] = default_platform_rev if default_platform_rev is not None else pd.NA
+            new_row_data['workload_repository_revision'] = default_workload_rev if default_workload_rev is not None else pd.NA
+
+            new_rows_list.append(new_row_data)
+    else:
+        logger.info("No new cluster_names found in source to add to target.")
+
+    if new_rows_list:
+        new_rows_df = pd.DataFrame(new_rows_list)
+        # Ensure new_rows_df has exactly the EXPECTED_TARGET_COLUMNS in the correct order
```

Comment:
It's possible there are other columns in there, and we just want to add without disrupting other values in there.  Maybe I'm worrying about something that isn't a problem, but....just a thought.